src/pages/Graph/GraphRouteHandler.tsx

```diff
@@ -1,84 +1,80 @@
 import * as React from 'react';
 import { RouteComponentProps } from 'react-router-dom';
-import { connect } from 'react-redux';
 
 import { GraphParamsType, GraphType, NodeParamsType, NodeType } from '../../types/Graph';
 import { EdgeLabelMode } from '../../types/GraphFilter';
 import * as LayoutDictionary from '../../components/CytoscapeGraph/graphs/LayoutDictionary';
 import GraphPage from '../../containers/GraphPageContainer';
-import { makeNamespaceGraphUrlFromParams, makeNodeGraphUrlFromParams } from '../../components/Nav/NavUtils';
 import { config } from '../../config';
 import * as Enum from '../../utils/Enum';
-import { KialiAppState } from '../../store/Store';
 import { NamespaceActions } from '../../actions/NamespaceAction';
 import Namespace from '../../types/Namespace';
-import { JsonString } from '../../types/Common';
-import { activeNamespaceSelector, previousGraphStateSelector } from '../../store/Selectors';
-import { Dispatch } from 'redux';
-import { HistoryManager } from '../../app/History';
+import { store } from '../../store/ConfigStore';
 
 const URLSearchParams = require('url-search-params');
 
+/**
+ * GraphRouteHandler handles bookmarkability. It parses the URL and updates state to
+ * reflect information in the URL path variable and query parameters.  After rendering the
+ * graph reflecting the URL it's job is done, it should not react to internal state changes.
+ */
+
+// We hold non-redux-managed graph state (i.e. GraphParamsType) in session state
 const SESSION_KEY = 'graph-params';
 
-type GraphURLProps = {
-  namespace: string;
+// GraphURLPathProps holds path variable values.  Currenly all path variables are
+// relevant only to a node graph
+type GraphURLPathProps = {
   app: string;
+  namespace: string;
   service: string;
   version: string;
   workload: string;
-  duration: string;
-  graphType: string;
-  injectServiceNodes: string;
-  layout: string;
-};
-
-const mapStateToProps = (state: KialiAppState) => {
-  return {
-    activeNamespace: activeNamespaceSelector(state),
-    previousGraphState: previousGraphStateSelector(state)
-  };
 };
 
-const mapDispatchToProps = (dispatch: Dispatch<any>) => {
-  return {
-    setActiveNamespace: (namespace: Namespace) => {
-      dispatch(NamespaceActions.setActiveNamespace(namespace));
-    },
-    setPreviousGraphState: (graphState: JsonString) => {
-      dispatch(NamespaceActions.setPreviousGraphState(graphState));
-    }
-  };
+// GraphURLPathProps holds query param values.  It combines information held in redux state
+// and session state (no information should be managed in both).
+//
+// A note on 'keepState': This is a necessary evil to support the 'Graph' option in the left
```

Comment:
You say evil, so i assume this must be bad. 
I think one choice would be to create the left menu Graph url with the settings we already have in redux.

Reply:
@josejulio We may be able to do that but I'm not sure it's worth making "routes" try to be too smart.  We have state split between redux and session, and also I'm not sure state is actually initialized in both places prior to an initial click on 'Graph'.  For now I think this is OK.  I'll remove the word 'evil' and that will make it all better :)